rust/src/pgsql/pgsql.rs

```diff
@@ -343,14 +360,28 @@ impl PgsqlState {
                 Ok((rem, request)) => {
                     sc_app_layer_parser_trigger_raw_stream_reassembly(flow, Direction::ToServer as i32);
                     start = rem;
-                    if let Some(state) = PgsqlState::request_next_state(&request) {
+                    let (temp_state, tx_completed) = if let Some(state) = PgsqlState::request_next_state(&request) {
                         self.state_progress = state;
+                        // PostreSQL progress states can be represented as a finite state machine
+                        // After the connection phase, the backend/ server will be mostly waiting in a state of `ReadyForQuery`, unless it's processing some request.
+                        // When the frontend wants to cancel a request, it will send a CancelRequest message over a new connection - to which there won't be any responses.
+                        // If the frontend wants to terminate the connection, the backend won't send any confirmation after receiving a Terminate request.
+                        // A simplified finite state machine for PostgreSQL v3 can be found at:
+                        // https://samadhiweb.com/blog/2013.04.28.graphviz.postgresv3.html
+                        (state, Self::request_is_complete(state))
+                    } else {
+                        (self.state_progress, false)
```

Comment:
Thanks.

This is complex stuff and needs even more doc/polishing I guess :
Could you rename `state` to tell if it is a Tx progress/state or a PgsqlState progress ?

Could tx_completed be an option of some state ?